packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,268 @@
+   *
+   * default: `true`
+   */
+  usePortal?: boolean;
+
+  /**
+   * Specifies the size of the Modal.
+   *
+   * default: `default`
+   */
+  size?: ModalSize;
+
+  /**
+   * Specifies the title of the Modal.
+   *
+   */
+  title?: string;
+
+  /**
+   * Callback to change the active state of the Modal.
+   *
+   */
+  setActive?: (
+    active: boolean,
+  ) => void | React.Dispatch<SetStateAction<boolean>>;
+
+  /**
+   * Callback to determine whether or not Modal should close when user tries to close it.
+   *
+   */
+  modalShouldClose?: () => boolean;
+
+  /**
+   * className applied to root overlay div.
+   *
+   */
+  className?: string;
+}
+
+/**
+ * # Modal
+ *
+ *  Modals place content on top of main window.
+ *
+```
+<Modal
+  active
+  usePortal
+  size="large"
+  setActive={setActive}
+  title="My Modal"
+  modalShouldClose={() => console.log('Modal is closing now!')}
+  >  
+  Modal content!
+</Modal>
+```
+ * @param props.active Boolean to describe whether or not Modal is active.
+ * @param props.usePortal Boolean to describe if content should be portaled to end of DOM, or appear in React DOM tree.
+ * @param props.size String to determine size of Modal. ['xxsmall', 'xsmall', 'small', 'default', 'large', 'xlarge']
+ * @param props.setActive Callback to change the active state of Modal.
+ * @param props.children Content to appear inside of Modal container.
+ * @param props.title Title for the Modal, will appear inside header tags.
+ * @param props.modalShouldClose Callback to determine whether or not Modal should close when user tries to close it.
+ * @param props.className className applied to overlay div.
+ *
+ */
+function Modal({
+  active = false,
+  usePortal = true,
+  size = ModalSize.Default,
+  setActive,
+  children,
+  title,
+  modalShouldClose,
+  className,
+  ...rest
+}: ModalProps) {
+  if (!active) {
+    return null;
+  }
+
+  const handleClose = () => {
+    // Don't close modal if modalShouldClose returns false or no setActive callback is passed
+    if (!setActive || (modalShouldClose && !modalShouldClose())) {
+      return;
+    }
+
+    setActive(false);
+  };
+
+  const handleEscape = useCallback((e: KeyboardEvent) => {
+    if (e.keyCode === EscapeKey) {
+      handleClose();
+    }
+  }, []);
+
+  useEventListener('keydown', handleEscape, {
+    options: { once: true },
+  });
+
+  const Root = usePortal ? Portal : Fragment;
+
+  return (
+    <Root>
+      <div {...rest} className={overlayStyle} onClick={handleClose}></div>
+      <div
+        className={cx(modalContentStyle, modalSizes[size], className)}
+        tabIndex={-1}
+      >
+        <Icon
+          glyph="X"
+          fill={'#5D6C74'}
+          size={Size.Large}
+          onClick={handleClose}
+          className={closeButton}
+          data-dismiss="modal"
+          aria-hidden="true"
+        />
+
+        {title && (
+          <header>
+            <h3 className={titleStyle}>{title}</h3>
```

Comment:
I think we should remove these for a couple reasons:

1. We haven't gotten to defining this pattern well yet. I think that it's fine to remove this until we have a well established pattern.
2. Philosophically, these components should be individually minimal, and this sort of thing should be composed of multiple components rather than being built in.

Reply:
This is coming from MMS, thoughts on this @hswolff?

Reply:
Agreed with @DesignerDave, let's omit from this primitive. It's too prescriptive to belong in the base Modal component.

Reply:
👍